activation/post.go

```diff
@@ -223,55 +246,49 @@ func (mgr *PostSetupManager) StartSession(ctx context.Context, opts PostSetupOpt
 	mgr.logger.With().Info("post setup completed",
 		log.String("node_id", mgr.id.String()),
 		log.String("commitment_atx", mgr.commitmentAtxId.String()),
-		log.String("data_dir", opts.DataDir),
-		log.String("num_units", fmt.Sprintf("%d", opts.NumUnits)),
+		log.String("data_dir", cfg.opts.DataDir),
+		log.String("num_units", fmt.Sprintf("%d", cfg.opts.NumUnits)),
 		log.String("labels_per_unit", fmt.Sprintf("%d", mgr.cfg.LabelsPerUnit)),
-		log.String("provider", fmt.Sprintf("%d", opts.ComputeProviderID)),
+		log.String("provider", fmt.Sprintf("%d", cfg.opts.ComputeProviderID)),
 	)
 	mgr.state = PostSetupStateComplete
 	return nil
 }
 
-func (mgr *PostSetupManager) prepareInitializer(ctx context.Context, opts PostSetupOpts) error {
+// InitializerConfig builds the config to be used by initialization in calls to
+// StartSession. Having this function be separate from StartSession provides a
+// means to understand if the post configuration is valid before kicking off a
+// very long running task (StartSession can take hours to complete).
+func (mgr *PostSetupManager) InitializerConfig(ctx context.Context, opts PostSetupOpts) (*InitializationConfig, error) {
```

Comment:
I find the name of this method a little misleading. It not only builds the config for the initializer. It creates the initializer.

Reply:
I see it similarly. I think this method could just be a "validate" method that only returns an error if the provided `PostSetupOpts` are invalid.

Instantiating a new initializer imo is part of `StartSession` and the CommitmentATX can just be added to the `PostSetupOpts` (and is selected like now if not set explicitly - imo there is nothing wrong with giving the user the option to select a commitmentATX of their liking).

Reply:
Its tricky. `InitializationConfig` does contain an initializer but it also has the atx id and opts.So if I called it an initializer you could argue that it's not because it has options as well.

It would be nice if I didn't have to update state in the builder and it could just be the initializer, but there seem to be a number of methods that access the bits of state in the builder that are updated.

What would you name it?

Reply:
Also see - https://github.com/spacemeshos/go-spacemesh/pull/4319/files/b14c55ae32e6de102c014cb0f164ededec95828f#r1175539280 for an explanation of why we have an instance of an initializer.

Reply:
Regarding naming: I would prefer `NewSession` that returns a `PostSession` that is then passed to `StartSession` 🙂